File: Data Structures & Algorithms/three-integer-sum/submission-3.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result = set()
        nums.sort()
        
        for i in range(len(nums)):
            seen = set() 
            for j in range(i + 1, len(nums)): 
                complement = -nums[i] - nums[j]
                if complement in seen: 
                    result.add(tuple([complement, nums[i], nums[j]]))
                seen.add(nums[j])


        return [list(i) for i in result]



        # for i in range(len(nums)): 
            # complement = target - nums[i]
            # if nums in seen: return indices
            # put complement in seen 

        

File: Data Structures & Algorithms/two-integer-sum/submission-3.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen = HashMap::new(); 

        for (i, &num) in nums.iter().enumerate(){ 
            let complement = target - num; 

            if let Some(&index) = seen.get(&complement){ 
                return vec![index as i32, i as i32];
            }
            seen.insert(num, i);
        }

        vec![]
    }
}
